server/src/main/java/org/candlepin/bind/BindContext.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.bind;
+
+import com.google.inject.Inject;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.EntitlementCertificate;
+import org.candlepin.model.Pool;
+import org.candlepin.model.PoolCurator;
+import org.candlepin.model.PoolQuantity;
+import org.candlepin.policy.EntitlementRefusedException;
+import org.candlepin.util.Util;
+
+import org.xnap.commons.i18n.I18n;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Container class for holding binding information
+ */
+public class BindContext {
+
+    private Consumer consumer;
+    private Consumer lockedConsumer;
+    private Map<String, PoolQuantity> poolQuantities;
+    private Map<String, Entitlement> entitlementMap;
+    private Map<String, EntitlementCertificate> certs;
+    private Map<String, Integer> quantities;
+
+    public boolean isQuantityRequested() {
+        return quantityRequested;
+    }
+
+    private boolean quantityRequested = false;
+
+    //change to generic type in future if needed
+    private EntitlementRefusedException exception;
+    private PoolCurator poolCurator;
+    private ConsumerCurator consumerCurator;
+    private I18n i18n;
+
+    public static BindContext build(BindContext context, Consumer consumer, Map<String, Integer> quantities) {
+        context.consumer = consumer;
+        context.quantities = quantities;
+        return context;
+    }
+
+    public BindContext(PoolCurator poolCurator, ConsumerCurator consumerCurator, I18n i18n) {
+        this.poolCurator = poolCurator;
+        this.consumerCurator = consumerCurator;
+        this.i18n = i18n;
+    }
+
+    public Consumer getConsumer() {
+        return consumer;
+    }
+
+    public Map<String, PoolQuantity> getPoolQuantities() {
+        if (poolQuantities == null) {
+            poolQuantities = new HashMap<String, PoolQuantity>();
+            List<Pool> pools = poolCurator.listAllByIds(quantities.keySet()).list();
```

Comment:
If there's any concern that the list returned could be huge, it might be worth considering not calling .list() straight away. None of the operations on the pools hit the DB again, so it's safe to use the cursor iteration here.